bin/make_splici_fasta.R

```diff
@@ -0,0 +1,157 @@
+# regions with spliced transcripts and introns
+
+# This script is also used to generate the 3 column spliced_intron.tx2gene_3col.tsv needed for alevin-fry.
+
+# finally, a list of mitochondrial genes is output for QC
+
+# load needed packages
+library(Biostrings)
+library(BSgenome)
+library(eisaR)
+library(GenomicFeatures)
+library(magrittr)
+library(optparse)
+library(tidyverse)
+
+# Set up optparse options
+option_list <- list(
+  make_option(
+    opt_str = c("-f", "--gtf"),
+    type = "character",
+    default = "annotation/Homo_sapiens.GRCh38.104.gtf.gz",
+    help = "File path for input gtf file",
+  ),
+  make_option(
+    opt_str = c("-g", "--genome"),
+    type = "character",
+    default = "fasta/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz",
+    help = "File path for reference fasta file",
+  ), 
+  make_option(
+    opt_str = c("-a", "--annotation_output"),
+    type = "character",
+    default = "annotation",
+    help = "Directory to write output gtf, tx2gene, and mitochondrial gene list files",
+  ), 
+  make_option(
+    opt_str = c("-o", "--fasta_output"),
+    type = "character",
+    default = "fasta",
+    help = "Directory to write output transcript fasta files",
+  ),
+  make_option(
+    opt_str = c("-s", "--assembly"),
+    type = "character",
+    default = "",
+    help = "Prefix name containing organism and ensembl assembly version to be used for file naming"
+  )
+)
+
+# Parse options
+opt <- parse_args(OptionParser(option_list = option_list))
+
+## files with spliced + intron regions
+spliced_intron_fasta_file <- paste0(opt$assembly, ".spliced_intron.txome.fa.gz")
+spliced_intron_gtf_file <- paste0(opt$assembly, ".spliced_intron.txome.gtf")
+spliced_intron_tx2gene_file <- paste0(opt$assembly, ".spliced_intron.tx2gene.tsv")
+spliced_intron_tx2gene_3col_file <- paste0(opt$assembly, ".spliced_intron.tx2gene_3col.tsv")
+
+mito_file <- paste0(opt$assembly, ".mitogenes.txt")
+
+# make final output file names needed
+
+spliced_intron_fasta <- file.path(opt$fasta_output, spliced_intron_fasta_file)
+spliced_intron_gtf <- file.path(opt$annotation_output, spliced_intron_gtf_file)
+spliced_intron_tx2gene <- file.path(opt$annotation_output, 
+                                    spliced_intron_tx2gene_file)
+
+spliced_intron_tx2gene_3col <- file.path(opt$annotation_output, 
+                                 spliced_intron_tx2gene_3col_file)
+
+mito_out <- file.path(opt$annotation_output, mito_file)
+
+# Check for output directory 
+if (!dir.exists(opt$fasta_output)) {
+  dir.create(opt$fasta_output)
+}
+
+if (!dir.exists(opt$annotation_output)) {
+  dir.create(opt$annotation_output)
+}
+
+
+# extract GRanges object containing genomic coordinates of each annotated transcript
+# both spliced transcripts and intronic regions
+grl <- eisaR::getFeatureRanges(
+  gtf = opt$gtf,
+  featureType = c("spliced", "intron"), 
+  flankLength = 86L,
```

Comment:
Do we want to make this an option? There was some discussion in the Alevin paper about choice of flank length, and it doesn't seem to matter much, but it seems reasonable to me to make it flexible. (Make sure to set `type= "integer"` when adding to options).
```suggestion
  flankLength = opts$flank_length,
```

Reply:
yes I was also reading that and thinking that having that flexibility could be useful. I'll go ahead and add to the options but set the default to 86.